### ISSUE TITLE
Maestro++ INT deployment failing

### ISSUE THREAD

**Issue by mmitche** [dotnet/arcade#4356]
The INT deployment is failing after the change to use stages rather than the release pipeline:

https://dev.azure.com/dnceng/internal/_build/results?buildId=426985&view=logs&s=d4a9a205-d52e-57fb-7b17-15a9d984af62&j=b277f1ae-44e7-5ce4-4818-00049b2c7654

It says the cluster is already in an ugprading state.

**Comment by ChadNedzlek** [comment]
Yup. Trying to figure out what's happening here. There is some DI error in the SF. But helpfully the useful part of the message is truncated in the logs, so I don't know what is missing or what's asking for it, so I'm trying to get a repro locally.

**Comment by sunandabalu** [comment]
@ChadNedzlek While you are at it, could you add the deploy stage to depend on the validate stage instead of the build? It does not make sense to deploy before Nuget and signing validation.

**Comment by ChadNedzlek** [comment]
The validate stage is magic, and not in our yaml.  I'm not really comfortable depending on a stage that isn't part of our definition, it will make us uniquely fragile to underlying changes.

**Comment by sunandabalu** [comment]
> The validate stage is magic, and not in our yaml. I'm not really comfortable depending on a stage that isn't part of our definition, it will make us uniquely fragile to underlying changes.

Its not magic, it comes from [post-build.yml] (https://github.com/dotnet/arcade-services/blob/c632da9bf0cfeb0ac1285488721957d2c6686c7c/eng/common/templates/post-build/post-build.yml#L25) :) And it gets added to all repos, that is the mandate that we have given to all repos. And the idea is to only deploy if Post-Build validation passes.

**Comment by ChadNedzlek** [comment]
Because it's in "common" it's not controlled by this repository, so it definitely is magic. This will mean that arcade updates will break our deployments, because obviously PR's don't do deployments.

If we want to try to solve that, that's fine, but that's not part of "make the build work", so we should open another issue.

**Comment by sunandabalu** [comment]
Fair, tracking the work here - https://github.com/dotnet/arcade/issues/4363

**Comment by sunandabalu** [comment]
@ChadNedzlek looks like the deploy yaml is referencing the wrong variable name for [MeastroEndPoint](https://github.com/dotnet/arcade-services/blob/1667b217002907ca8b816be0806571b450f15385/eng/deploy.yaml#L155)
It should be using [this](https://github.com/dotnet/arcade-services/blob/1667b217002907ca8b816be0806571b450f15385/eng/deploy.yaml#L25)

**Comment by ChadNedzlek** [comment]
Darn it. I messed this one up because the thing needs 2 endpoints... a test one and a publish one, which change from environment to environment, so one "MaestroUrl" didn't work. :-|

**Comment by ChadNedzlek** [comment]
I think I've fixed this. I'll see if the next deployment works.

**Comment by riarenas** [comment]
The new pipeline is working all the way through the scenario tests now that we have fixed all of them, but the "deploy darc" stage is failing.

https://dev.azure.com/dnceng/internal/_build/results?buildId=433543&view=results

@mmitche I think we can remove that step as we are now publishing it during the build as part of the publishing stages?

**Comment by mmitche** [comment]
Yeah we can remove that

**Comment by riarenas** [comment]
Removed the stage in https://github.com/dotnet/arcade-services/pull/797 and https://dev.azure.com/dnceng/internal/_build/results?buildId=434380&view=results is green!

I think we can close this issue now if everything looks right to you @ChadNedzlek

**Comment by ChadNedzlek** [comment]
Honestly I don't know much about this issue. I just copied the existing releases into YAML because I needed to add a deployment stage and I detest increasing our debt by putting it in the old way, so I migrated everything to the new way to unblock myself. :-)

If it's green and my service exists (it does) then I'm happy.
